### PR TITLE
[CI] Add debugging and fix some issue when detecting if it is a VM

### PR DESCRIPTION
**Pull request by mandel-macaque**
Small useful changes:

1. Simpler grep.
2. VMware not VMWare
3. Some debugging in case if fails again.

### CODE REVIEW

File: tools/devops/automation/templates/mac/build.yml [hunk 1]
```diff
@@ -161,6 +172,9 @@ steps:
     $testsPath = "$(Build.SourcesDirectory)/artifacts/mac-test-package/tests"
     Write-Host "Tests path is $testsPath"
 
+    # pring enviroment
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/mac/build.yml:175]:
```suggestion
    # print enviroment
```

**Review by rolfbjarne — APPROVED** [review]

File: tools/devops/automation/templates/mac/build.yml [hunk 2]
```diff
@@ -42,16 +42,27 @@ steps:
   displayName: 'Dump Hardware'
 
 - bash: |
-    vmware=$(ioreg -l | grep -e Manufacturer -e 'Vendor Name' | grep -e 'VMWare' | wc -l)
+    vmware=$(ioreg -l | grep -e 'VMware' | wc -l)
+    echo "VMWare count $vmware"
```
Comment (spouliot) [inline tools/devops/automation/templates/mac/build.yml:46]: `VMware not VMWare`
avoid future c&p typos ;-)

File: tools/devops/automation/templates/mac/build.yml [hunk 3]
```diff
@@ -42,16 +42,27 @@ steps:
...
     if [[ $vmare -gt 0 ]]; then
+      echo "Using VMWare virtualization."
```
Comment (spouliot) [inline tools/devops/automation/templates/mac/build.yml:48]: VMware not VMWare

File: tools/devops/automation/templates/mac/build.yml [hunk 4]
```diff
@@ -42,16 +42,27 @@ steps:
...
       echo "##vso[task.setvariable variable=VM_VENDOR]VMWare"
```
Comment (spouliot) [inline tools/devops/automation/templates/mac/build.yml:49]: VMware not VMWare

File: tools/devops/automation/templates/mac/build.yml [hunk 5]
```diff
@@ -42,16 +42,27 @@ steps:
...
+    echo "VirtualBox count $virtualbox"
     if [[ $virtualbox -gt 0 ]]; then
+      echo "Using VirtualBox virtualization."
       echo "##vso[task.setvariable variable=VM_VENDOR]VirtualBox"
     fi
 
+    parallels=$(ioreg -l | grep -e 'Parallels' | wc -l)
+    echo "Parallels count $parallels"
+    if [[ $parallels -gt 0 ]]; then
+      echo "Using VMWare virtualization."
```
Comment (spouliot) [inline tools/devops/automation/templates/mac/build.yml:62]: VMware not VMWare
but maybe you meant `Parallels` ?

**Review by spouliot — CHANGES_REQUESTED** [review]

File: tools/devops/automation/templates/mac/build.yml (hunk 2 above)
Comment (mandel-macaque) [inline tools/devops/automation/templates/mac/build.yml:46]:
```suggestion
    echo "VMware count $vmware"
```

File: tools/devops/automation/templates/mac/build.yml (hunk 3 above)
Comment (mandel-macaque) [inline tools/devops/automation/templates/mac/build.yml:48]:
```suggestion
      echo "Using VMware virtualization."
```

File: tools/devops/automation/templates/mac/build.yml (hunk 4 above)
Comment (mandel-macaque) [inline tools/devops/automation/templates/mac/build.yml:49]:
```suggestion
      echo "##vso[task.setvariable variable=VM_VENDOR]VMware"
```

File: tools/devops/automation/templates/mac/build.yml (hunk 5 above)
Comment (mandel-macaque) [reply tools/devops/automation/templates/mac/build.yml:62]: true, ups
```suggestion
      echo "Using Parallels virtualization."
```

**Review by spouliot — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Review by stephen-hawley — APPROVED** [review]

**Pull request merged**